frontend/src/app/testResults/DownloadResultsCsvModal.tsx

```diff
@@ -60,33 +59,37 @@ export const DownloadResultsCsvModal = ({
     ...filterParams,
   };
 
-  const getResults = useImperativeQuery<GetFacilityResultsForCsvWithCountQuery>(
-    GetFacilityResultsForCsvWithCountDocument,
-    {
+  const [downloadTestResultsQuery, { loading }] =
+    useGetFacilityResultsForCsvWithCountLazyQuery({
+      variables,
       fetchPolicy: "no-cache",
-    }
-  );
+      onCompleted: (data: GetFacilityResultsForCsvWithCountQuery) =>
+        handleComplete(data),
+      onError: (error: ApolloError) => handleError(error),
+    });
 
-  const downloadResults = async () => {
-    const { data, error } = await getResults(variables);
-    if (error) {
-      showError("Error downloading results", error.message);
-      setLoading(false);
-    } else if (data.testResultsPage && data.testResultsPage.content) {
+  const handleComplete = (data: GetFacilityResultsForCsvWithCountQuery) => {
+    if (data.testResultsPage && data.testResultsPage.content) {
       const csvResults = parseDataForCSV(
         data.testResultsPage.content,
         multiplexEnabled
       );
       setResults(csvResults);
-      setLoading(false);
-      csvLink?.current?.link.click();
-      closeModal();
     } else {
       showError("Unknown error downloading results");
-      setLoading(false);
     }
   };
 
+  const handleError = (error: ApolloError) => {
+    showError("Error downloading results", error.message);
```

Comment:
I know this was in the original code so not a regression, but this would pop up a user-facing toast - can we confirm with @lauraykerr on content? I'm not sure but I don't think we want to surface the message from the ApolloError since it will likely not be super user-friendly